src/iris_grib/_load_convert.py

```diff
@@ -209,7 +209,7 @@ def _masker(item):
 # Non-standardised usage for negative forecast times.
 def _hindcast_fix(forecast_time):
     """Return a forecast time interpreted as a possibly negative value."""
-    uft = np.uint32(forecast_time)
+    uft = np.array(forecast_time).astype(np.uint32)
```

Comment:
I'm gonna merge this anyway, but is there a particular reason you didn't do:

```suggestion
    uft = np.array(forecast_time, dtype=np.uint32)
```

Reply:
Recent versions of numpy (<2) were warning that the silent wrapping of values too big or small for the stated dtype was effectively deprecated + going to raise an error.
Which does happen with numpy2.

The 'astype' is a fix suggested by those warnings, which makes this code compatible with numpy 2.

**However** what went missing here is that this new code fails the tests when run under numpy 2, so further changes are still required to unpin our locks...